code/game/objects/items/devices/megaphone.dm

```diff
@@ -27,7 +27,7 @@
 		return
 
 	var/message = copytext(sanitize(input(user, "Shout a message?", "Megaphone", null)  as text),1,MAX_MESSAGE_LEN)
-	if(!message)
+	if(!message || message == "")
```

Comment:
@RemieRichards Sorry, I meant to check for an empty message, not a string, to catch as well when a player just 'speaks' some spaces, but that's not really necessary